pkg/kotsclient/cluster_list.go

```diff
@@ -5,22 +5,39 @@ import (
 	"net/http"
 	"strconv"
 
+	"github.com/pkg/errors"
+
 	"github.com/replicatedhq/replicated/pkg/types"
 )
 
 type ListClustersRequest struct {
 }
 
 type ListClustersResponse struct {
-	Clusters []*types.Cluster `json:"clusters"`
+	Clusters      []*types.Cluster `json:"clusters"`
+	TotalClusters int              `json:"totalClusters"`
 }
 
-func (c *VendorV3Client) ListClusters(includeTerminated bool) ([]*types.Cluster, error) {
+func (c *VendorV3Client) ListClusters(includeTerminated bool, startTime string, endTime string) ([]*types.Cluster, error) {
 	reqBody := &ListClustersRequest{}
-	clusters := ListClustersResponse{}
-	err := c.DoJSON("GET", fmt.Sprintf("/v3/clusters?include-terminated=%s", strconv.FormatBool(includeTerminated)), http.StatusOK, reqBody, &clusters)
-	if err != nil {
-		return nil, err
+
+	allClusters := []*types.Cluster{}
+	page := 0
+	for {
+		clusters := ListClustersResponse{}
+		err := c.DoJSON("GET", fmt.Sprintf("/v3/clusters?currentPage=%d&show-terminated=%s&start-time=%s&end-time=%s", page, strconv.FormatBool(includeTerminated), startTime, endTime),
```

Comment:
Looking at server side implementation, timestamps can have spaces.  If that's going to be supported, URL parameters have to be encoded.  Use url.Values instead of Sprintf like in [this function](https://github.com/replicatedhq/replicated/blob/b746cff945b1cb5d0e1f3c2f0563e855cc0b2182/pkg/kotsclient/channel.go#L19-L25).